src/main/kotlin/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtSharedTableBackupManager.kt

```diff
@@ -0,0 +1,456 @@
+                    return ignoreClassesOfCreateTableRequest.contains(clazz.name)
+                }
+
+                override fun shouldSkipField(field: FieldAttributes): Boolean {
+                    return false
+                }
+            }).enableComplexMapKeySerialization().create()
+    val ignoreClassesOfCreateTableRequest: Set<String> = ImmutableSet.of(
+            "com.amazonaws.RequestClientOptions",
+            "com.amazonaws.event.ProgressListener"
+    )
+    val charset = Charset.forName("utf-8")
+
+    override fun createBackup(createBackupRequest: CreateBackupRequest): MtBackupMetadata {
+
+        val backup = getBackup(createBackupRequest.backupName)
+        if (backup != null) {
+            throw MtBackupException("Backup with that ID already exists: $backup")
+        } else {
+            val startTime = System.currentTimeMillis()
+            val virtualMetadata = backupVirtualTableMetadata(createBackupRequest)
+            val tenantTableCounts: Map<TenantTableBackupMetadata, Long> = virtualMetadata
+                    .map { metadata -> TenantTableBackupMetadata(
+                            createBackupRequest.backupName,
+                            metadata.tenantName,
+                            metadata.createTableRequest.tableName) }
+                    .associateBy({ it }, { 0L })
+            val newMetadata = MtBackupMetadata(createBackupRequest.backupName,
+                    Status.IN_PROGRESS, tenantTableCounts, startTime)
+
+            commitBackupMetadata(newMetadata)
+            return getBackup(newMetadata.mtBackupName)!!
+        }
+    }
+
+    open fun backupVirtualTableMetadata(
+        createBackupRequest: CreateBackupRequest
+    ): List<MtTableDescriptionRepo.MtCreateTableRequest> {
+        val startTime = System.currentTimeMillis()
+        // write out table metadata
+        val startKey: MtTableDescriptionRepo.MtCreateTableRequest? = null
+        val batchSize = 100
+        var tenantTableCount = 0
+        val tenantTables = Lists.newArrayList<MtTableDescriptionRepo.MtCreateTableRequest>()
+        do {
+            val listTenantMetadataResult =
+                    sharedTableMtDynamo.mtTableDescriptionRepo.listVirtualTableMetadata(
+                            MtTableDescriptionRepo.ListMetadataRequest()
+                                .withExclusiveStartKey(startKey)
+                                .withLimit(batchSize))
+            commitTenantTableMetadata(createBackupRequest.backupName, listTenantMetadataResult)
+            tenantTables.addAll(listTenantMetadataResult.createTableRequests)
+            tenantTableCount += listTenantMetadataResult.createTableRequests.size
+        } while (listTenantMetadataResult.lastEvaluatedTable != null)
+        logger.info("${createBackupRequest.backupName}: Finished generating backup metadata for " +
+                "${tenantTables.size} virtual table in ${System.currentTimeMillis() - startTime} ms")
```

Comment:
Maybe emit Micrometer metrics for duration?

Reply:
sure, will include in a followup